pkg/strategy/fmaker/strategy.go

```diff
@@ -0,0 +1,532 @@
+
+		log.Info(s.Position.GetBase().Mul(kline.Close).Float64(), AskAlphaBoundary, er, BidAlphaBoundary)
+
+		BidPrice := kline.Close.Mul(fixedpoint.One.Sub(s.Spread))
+		BidQty := s.QuantityOrAmount.CalculateQuantity(BidPrice)
+		BidQty = BidQty //.Mul(fixedpoint.One.Add(fixedpoint.NewFromFloat(eq)))
+
+		AskPrice := kline.Close.Mul(fixedpoint.One.Add(s.Spread))
+		AskQty := s.QuantityOrAmount.CalculateQuantity(AskPrice)
+		AskQty = AskQty //.Mul(fixedpoint.One.Add(fixedpoint.NewFromFloat(eq)))
+
+		if er > 0 || (er < 0 && er > AskAlphaBoundary/kline.Close.Float64()) {
+			submitOrder := types.SubmitOrder{
+				Symbol:   s.Symbol,
+				Side:     types.SideTypeBuy,
+				Type:     types.OrderTypeLimitMaker,
+				Price:    BidPrice,
+				Quantity: BidQty, //0.0005
+			}
+			createdOrders, err := orderExecutor.SubmitOrders(ctx, submitOrder)
+			if err != nil {
+				log.WithError(err).Errorf("can not place orders")
+			}
+			s.orderStore.Add(createdOrders...)
+			s.activeMakerOrders.Add(createdOrders...)
+			s.tradeCollector.Process()
+
+			//submitOrder = types.SubmitOrder{
+			//	Symbol:   s.Symbol,
+			//	Side:     types.SideTypeSell,
+			//	Type:     types.OrderTypeLimitMaker,
+			//	Price:    kline.Close.Mul(fixedpoint.One.Add(s.Spread)),
+			//	Quantity: fixedpoint.NewFromFloat(math.Max(math.Min(eq, 0.003), 0.0005)), //0.0005
+			//}
+			//createdOrders, err = orderExecutor.SubmitOrders(ctx, submitOrder)
+			//if err != nil {
+			//	log.WithError(err).Errorf("can not place orders")
+			//}
+			//s.orderStore.Add(createdOrders...)
+			//s.activeMakerOrders.Add(createdOrders...)
+			//s.tradeCollector.Process()
+		}
+		if er < 0 || (er > 0 && er < BidAlphaBoundary/kline.Close.Float64()) {
+			submitOrder := types.SubmitOrder{
+				Symbol:   s.Symbol,
+				Side:     types.SideTypeSell,
+				Type:     types.OrderTypeLimitMaker,
+				Price:    AskPrice,
+				Quantity: AskQty, //0.0005
+			}
+			createdOrders, err := orderExecutor.SubmitOrders(ctx, submitOrder)
+			if err != nil {
+				log.WithError(err).Errorf("can not place orders")
+			}
+			s.orderStore.Add(createdOrders...)
+			s.activeMakerOrders.Add(createdOrders...)
+			s.tradeCollector.Process()
+
+			//submitOrder = types.SubmitOrder{
+			//	Symbol:   s.Symbol,
+			//	Side:     types.SideTypeBuy,
+			//	Type:     types.OrderTypeLimitMaker,
+			//	Price:    kline.Close.Mul(fixedpoint.One.Sub(s.Spread)),
+			//	Quantity: fixedpoint.NewFromFloat(math.Max(math.Min(eq, 0.003), 0.0005)), //0.0005
+			//}
+			//createdOrders, err = orderExecutor.SubmitOrders(ctx, submitOrder)
+			//if err != nil {
+			//	log.WithError(err).Errorf("can not place orders")
+			//}
+			//s.orderStore.Add(createdOrders...)
+			//s.activeMakerOrders.Add(createdOrders...)
+			//s.tradeCollector.Process()
+		}
+
+	})
+
+	return nil
+}
+
+func tanh(x float64) float64 {
+	y := (math.Exp(x) - math.Exp(-x)) / (math.Exp(x) + math.Exp(-x))
+	return y
+}
+
+func mean(xs []float64) float64 {
```

Comment:
types.Float64Slice (i.e., []float64)
Is also a type of Series.
You could call types.Mean instead